components/sup/src/manager/service/spec.rs

```diff
@@ -71,6 +75,23 @@ impl FromStr for DesiredState {
     }
 }
 
+/// Helper enum to abstract over spec type.
```

Comment:
This comment feels superfluous since this abstracting over `X` type is what all enums do. For someone who doesn't already know, a helpful comment might be something that describes what a `Spec` means or is used for.

Reply:
These weren't my comments (https://github.com/habitat-sh/habitat/blob/5ec083ff17a4fbfdb2506baf341e587238bd8bea/components/sup/src/main.rs#L918-L936) so I didn't want to remove what somebody else thought was helpful. Happy to remove it though!

Reply:
@reset I wrote them initially; feel free to edit / remove as you wish :smile: